### PR TITLE
feat(alerts): Create eap snuba subscription using rpc timeseries request

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Updates Alert Rule creation for EAP alerts to use the Create Subscription RPC endpoint in snuba (https://github.com/getsentry/snuba/pull/6499)

Constructs and passes a `TimeSeriesRequest` to be used in the subscription creation.

### CODE REVIEW

File: src/sentry/utils/snuba_rpc.py [hunk 1]
```diff
@@ -99,24 +103,31 @@ def rpc(req: SnubaRPCRequest, resp_type: type[RPCResponseType]) -> RPCResponseTy
 
 
 def _make_rpc_request(
-    endpoint_name: str, class_version: str, req: SnubaRPCRequest
+    endpoint_name: str,
+    class_version: str,
+    req: SnubaRPCRequest | CreateSubscriptionRequest,
```
Comment (edwardgou-sentry) [inline src/sentry/utils/snuba_rpc.py:108]: Currently, the `CreateSubscriptionRequest` protobuf doesn't have a `meta` attribute with a `referrer`. I've added `CreateSubscriptionRequest` as a valid type here to handle these scenarios. I'm open to updating the protobuf to include `meta` instead if we feel that would be better

File: src/sentry/snuba/tasks.py [hunk 2]
```diff
@@ -261,6 +284,24 @@ def _create_snql_in_snuba(subscription, snuba_query, snql_query, entity_subscrip
...
+    subscription, snuba_query, rpc_time_series_request: TimeSeriesRequest, entity_subscription
+):
+    subscription_request = CreateSubscriptionRequest(
+        time_series_request=rpc_time_series_request,
+        time_window_secs=snuba_query.time_window,
+        resolution_secs=snuba_query.resolution,
+    )
+
+    try:
+        response = snuba_rpc.rpc(subscription_request, CreateSubscriptionResponse)
```
Comment (wmak) [inline src/sentry/snuba/tasks.py:297]: Could you add a `snuba_rpc.create_subscription` method instead?

File: src/sentry/utils/snuba_rpc.py (hunk 1 above)
Comment (wmak) [reply src/sentry/utils/snuba_rpc.py:108]: I think this is fine for now, I'll clean this up in a future pass :+1:

**Review by wmak — APPROVED** [review]
lgtm, but i'm not that familiar with the alerts code

**Pull request merged**